build.cmake

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env -S cmake -P
+
+# This cross-platform script automates the process of creating a build directory and
+# compiling MuseScore. Despite being written in the CMake language, you should think
+# of this file as like a shell script or batch file rather than as part of the CMake
+# project tree. As always, the project tree starts in the top-level CMakeLists.txt.
+
+if(NOT DEFINED CMAKE_SCRIPT_MODE_FILE)
+    file(RELATIVE_PATH SCRIPT_PATH "${CMAKE_BINARY_DIR}" "${CMAKE_CURRENT_LIST_FILE}")
+    message(FATAL_ERROR
+        "This file is a script. You should run it with:\n"
+        "  \$ cmake -P ${SCRIPT_PATH} [args...]\n"
+        "Don't try to use it in other CMake files with include().\n"
+        "See https://cmake.org/cmake/help/latest/manual/cmake.1.html#run-a-script"
+    )
+endif()
+
+# CMake arguments up to '-P' (ignore these)
+set(i "1")
+while(i LESS "${CMAKE_ARGC}")
+    if("${CMAKE_ARGV${i}}" STREQUAL "-P")
+        math(EXPR i "${i} + 1") # ignore '-P' option
+        break() # done with CMake arguments
+    endif()
+    math(EXPR i "${i} + 1") # next argument
+endwhile()
+
+# Script arguments (store these for later processing)
+# By convention, SCRIPT_ARG[0] is the name of the script
+set(SCRIPT_ARGV "") # empty argument list
+set(SCRIPT_ARGC 0) # number of arguments
+while(i LESS "${CMAKE_ARGC}")
+    list(APPEND SCRIPT_ARGV "${CMAKE_ARGV${i}}")
+    math(EXPR SCRIPT_ARGC "${SCRIPT_ARGC} + 1")
+    math(EXPR i "${i} + 1") # next argument
+endwhile()
+
+#### INTERESTING STUFF STARTS HERE ####
+
+# Default values for build variables
+set(SOURCE_PATH "${CMAKE_CURRENT_LIST_DIR}") # MuseScore root directory
+set(BUILD_PATH "${SOURCE_PATH}/builds/default")
+set(BUILD_TYPE "RelWithDebInfo")
+
+# CPUS
+cmake_host_system_information(RESULT CPUS QUERY NUMBER_OF_LOGICAL_CORES)
+if(NOT "${CPUS}" GREATER 0)
+    include(ProcessorCount)
+    ProcessorCount(CPUS)
+endif()
+
+# Qt Creator
+if(DEFINED ENV{QTDIR})
+    set(ENV{PATH} "$ENV{QTDIR}/bin:$ENV{PATH}") # Add selected build kit to PATH
+endif()
+
+# MSVC / Visual Studio
```

Comment:
What about QtCreator/MinGW, and QtCreator/MSVC?

Reply:
Not got that far yet